docs/src/proposals/transactions-v2.md

```diff
@@ -158,15 +157,73 @@ pub struct AddressMap {
   #[serde(with = "short_vec")]
   pub entries: Vec<u8>,
 }
+
+bitflags::bitflags! {
+    struct IxConfig: u8 {
+        const NONE = 0b00000000;
+        /// The signer privilege may be propagated to arbitrary
+        /// instruction depth by default. When this config flag
+        /// is enabled, the signer privilege cannot be propagated.
+        /// Only the signer privileges by the direct invoker
+        /// (program or user wallet) will be usable by the invoked
+        /// instruction.
+        const DISABLE_SIGNER_PROPAGATION = 0b00000001;
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct CompiledInstructionV2 {
+  /// NEW: Modify runtime behavior for this instruction
+  pub config: IxConfig,
+
+  /// unchanged
+  pub program_id_index: u8,
+
+  /// NEW: Only programs in this list may be invoked by this instruction
+  #[serde(with = "short_vec")]
+  pub invokable_program_id_indexes: Vec<u8>,
```

Comment:
This kinda but I guess not sufficiently covers the need for the bit flags

Reply:
The bit flag concept allows for a limited number of other flags/uses cases in the future, so helpful in general.  But for signer propagation, another possibility is to discriminately allow signer propagation by ordering the list of invokable ids and a count of how many of those should the signing be propagated to.  Kinda like ro vs rw accounts in the message.

Reply:
The motivation behind disabling signer propagation is more around ensuring that a program can't use the system or token program to drain my assets. Most times, the system and token programs going to be the programs which are invoked with signer propagation so I don't think that level of granularity that way adds much safety to users.